tests/kernel/timer/timer_behavior/pytest/test_timer.py

```diff
@@ -99,3 +99,6 @@ def test_flash(dut: DeviceAdapter, tool, tool_options, config,
         stats, stats_count = tool.run(seconds, tool_options)
         assert stats_count
         do_analysys(test, stats, stats_count, config, sys_clock_hw_cycles_per_sec)
+
+    # Let the running test's image output to be fully captured from device.
+    dut.readlines()
```

Comment:
This relies on an EOF condition on the stream, no?  What about things like UARTs that say open forever?  Or is there handling at another layer for detecting "end of test"?

Reply:
I'm kinda confused why is this needed at all: when running with the external tool, the output of the Zephyr application shouldn't matter at all, what matters is what was collected from the external tool.

Reply:
> This relies on an EOF condition on the stream, no? What about things like UARTs that say open forever? Or is there handling at another layer for detecting "end of test"?

in this case, it is either Twister Pytest `DeviceAdapter` and `Harness` responsibility, or the board-related setup also at the Twister layer to release the board on timeout.

Reply:
> I'm kinda confused why is this needed at all: when running with the external tool, the output of the Zephyr application shouldn't matter at all, what matters is what was collected from the external tool.

full Zephyr application test output matters for two reasons :
1) to have evidence the test is actually passed and the external data is valid;
2) to have the test app's statistics collected as well, which is interesting to analyze for history stats, and to compare with the timer test app variant which runs without the external tool on the sane board.